src/plutus/cek.md

```diff
@@ -0,0 +1,136 @@
+# The CEK Machine
+
+This page presents the operational semantics of the Plutus language using an abstract machine - a [CEK machine](https://en.wikipedia.org/wiki/CEK_Machine).
+Although more complex than traditional reduction semantics, the CEK machine offers an efficient evaluation strategy that enables precise modeling of resource usage and cost.
+The production Haskell evaluator is based on the CEK machine, and it also provides a practical foundation for alternative implementations.
+
+The following listing defines some key concepts of the CEK machine.
+
+```text
+Σ ∈ State ::= 𝑠; 𝜌 ⊳ 𝑀  // Computing M under environment 𝜌 with stack 𝑠
+            | 𝑠 ⊲ 𝑉     // Returning a value 𝑉 to stack 𝑠
+            | ⬥        // Throwing an error
+            | ◻ 𝑉      // Final state with result 𝑉
+
+𝑠 ∈ Stack ::= 𝑓*  // A stack has zero or more stack frames
+
+𝑉 ∈ CEK value ::= 〈con T 𝑐〉         // A constant 𝑐 with type T
+                | 〈delay 𝑀 𝜌〉      // A delayed computation, with an
+                                   // associated environment
+                | 〈lam 𝑥 𝑀 𝜌〉      // A lambda abstraction, with an
+                                   // associated environment
+                | 〈constr 𝑖 𝑉*〉   // A constructor application, where
+                                   // all arguments are values
+                | 〈builtin 𝑏 𝑉* 𝜂〉 // A builtin application with all supplied
+                                   // arguments as values, and expecting
+                                   // at least one more argument
+
+𝜌 ∈ Environment ::= []        // An empty environment
+                  | 𝜌[𝑥 ↦ 𝑉]  // Associate 𝑥 with 𝑉 in the environment
+
+𝜂 ∈ Expected builtin arguments ::= [𝜄]  // One argument
```

Comment:
I don't think it says anywhere what `𝜄` is.  Those are the things that appear in signatures of builtins (which appear in the tables of builtins in Chapter 4) and they're explained in Section 2.2.2.3 of the specification.  You probably don't want to get into the details of that though.

Reply:
Yeah, no need to get into the details. I'll just add that `𝜄` is either a term or a type argument.